haddock-api/src/Haddock/Interface.hs

```diff
@@ -165,6 +167,7 @@ createIfaces verbosity flags instIfaceMap mods = do
 processModule :: Verbosity -> ModSummary -> [Flag] -> IfaceMap -> InstIfaceMap -> Ghc (Maybe Interface)
 processModule verbosity modsum flags modMap instIfaceMap = do
   out verbosity verbose $ "Checking module " ++ moduleString (ms_mod modsum) ++ "..."
+  liftIO $ hSetEncoding stderr utf8
```

Comment:
@igrep I would say you should get help on cabals issue tracker: https://github.com/haskell/cabal/issues 

Also: Why would you want to bootstrap? An old version of cabal surely can compile cabal HEAD. So you don't need to take route via bootstrap.